tensorflow/core/kernels/string_split_op.cc

```diff
@@ -118,10 +117,50 @@ class StringSplitOp : public OpKernel {
     }
   }
 
- private:
+ protected:
   bool skip_empty_;
 };
 
+class StringSplitOp : public StringSplitOpBase {
+ public:
+  explicit StringSplitOp(OpKernelConstruction* context)
+      : StringSplitOpBase(context) {
+    skip_empty_ = true;
+    // By default skip_empty_ is true. We only get the value from attr if it is
+    // available, so that it is backward compatible.
+    context->GetAttr("skip_empty", &skip_empty_);
+  }
+
+  Status ValidateDelimiter(const string& delimiter) { return Status::OK(); }
+
+  Status Split(const string& input, const string& delimiter,
+               std::vector<string>* parts) {
+    *parts = SplitString(input, delimiter, skip_empty_);
+    return Status::OK();
+  }
+};
+
+class StringSplitUTF8Op : public StringSplitOpBase {
+ public:
+  explicit StringSplitUTF8Op(OpKernelConstruction* context)
+      : StringSplitOpBase(context) {
+    // The skip_empty is not optional (unlike StringSplitOp)
+    OP_REQUIRES_OK(context, context->GetAttr("skip_empty", &skip_empty_));
```

Comment:
I'm probably missing something, but I don't see `skip_empty_` being used by `StringSplitUTF8Op`. How is it having an effect?

Reply:
The PR has been updated with `skip_empty_`.